crates/ruma-client-api/src/sync/sync_events.rs

```diff
@@ -62,3 +72,184 @@ impl DeviceLists {
         self.changed.is_empty() && self.left.is_empty()
     }
 }
+
+/// A state event that has either the sync or stripped format.
+#[derive(Debug, Clone)]
+#[allow(clippy::exhaustive_enums)]
+pub enum AnySyncOrStrippedStateEvent {
+    /// An event using the sync format.
+    Sync(Box<AnySyncStateEvent>),
```

Comment:
Why the `Box`?

Reply:
Because [`clippy::large_enum_variant`](https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant) :stuck_out_tongue:

Reply:
For info IDE says that `Sync` variant size is 480 and `Stripped` variant size is 216.

Reply:
I don't like it, the ratio of theoretical perf improvement to annoyance seems too low. Can we just silence that lint here?

Reply:
Done.